reana/reana_dev/git.py

```diff
@@ -767,4 +740,92 @@ def _commit_and_publish_version_bumps(components):
+@click.option(
+    "--component",
+    "-c",
+    multiple=True,
+    default=["CLUSTER", "CLIENT"],
+    help="Which components? [shortname|name|.|CLUSTER|ALL]",
+)
+@click.option(
+    "--use-latest-known-tag",
+    is_flag=True,
+    default=False,
+    help="Should the upgrade use the latest known tag of the shared modules?"
+    " If so, the latest known tag of the shared modules will be used to upgrade the"
+    " provided components. Else, the upgrade will only occur if the latest commit of the"
+    " shared modules points at a tag, in other case, the command will be aborted.",
+)
+@click.option(
+    "--create-new-version-bump-commit",
+    is_flag=True,
+    default=False,
+    help="Should version bump changes be included in a separate commit?"
+    ' If so, a new commit with "installation: bump shared modules" message will'
+    " be created for the provided components. Else, changes will be integrated"
+    " in the latest commit of each component.",
+)
+@click.pass_context
+def git_upgrade_shared_modules(
+    ctx, component, use_latest_known_tag, create_new_version_bump_commit
+):  # noqa: D301
+    """Upgrade selected components to latest REANA-Commons/DB version.
+
+    \b
+    :param components: The option ``component`` can be repeated. The value may
+                       consist of:
+                         * (1) standard component name such as
+                               'reana-workflow-controller';
+                         * (2) short component name such as 'r-w-controller';
+                         * (3) special value '.' indicating component of the
+                               current working directory;
+                         * (4) special value 'CLUSTER' that will expand to
+                               cover all REANA cluster components [default];
+                         * (5) special value 'CLIENT' that will expand to
+                               cover all REANA client components;
+                         * (6) special value 'DEMO' that will expand
+                               to include several runable REANA demo examples;
+                         * (7) special value 'ALL' that will expand to include
+                               all REANA repositories.
+    :type component: str
+    """
+
+    def _create_commit_or_amend(components):
+        for component in components:
+            commit_cmd = "git commit --amend --no-edit"
+            if create_new_version_bump_commit:
+                commit_cmd = 'git commit -m "installation: bump shared modules"'
+
+            run_command(
+                f"git add setup.py requirements.txt && {commit_cmd}", component,
+            )
+
+    def _push_to_origin(components):
+        for component in components:
+            branch = run_command(
+                "git branch --show-current", component, return_output=True
+            )
+            run_command(
+                f"git push --force origin {branch}", component,
```

Comment:
I'm not fully sure about this part. If one is checking out a feature branch of a colleague, will this upgrade their PR?

Reply:
Nope, if you are checking only, you are in "latest" or "debug" mode, so you wouldn't need to upgrade minimal versions. The command to upgrade common modules is only used by the integrator, where it is OK to update the PR, because it saves time since the integrator won't need to ask people "please bump r-commons in the other PR in this set".  So I consider it as a "service provided by integrators so that developers don't have to do unnecessary bumping".